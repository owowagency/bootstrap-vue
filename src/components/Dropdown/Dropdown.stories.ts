import Dropdown from '.';
import DropdownMenu from '@/components/DropdownMenu';
import {sizes} from '@/composables/useSize';
import {variants} from '@/composables/useVariant';

export default {
    title: 'Components/Dropdown',
    component: Dropdown,
    argTypes: {},
};

const defaultArgs = {
    items: [
        {label: 'Item 1'},
        {label: 'Item 2'},
        {label: 'Item 3'},
    ],
    text: 'I am a dropdown',
};

const template = (args) => ({
    components: {Dropdown},
    setup: () => ({args}),
    template: '<Dropdown v-bind="args" />',
});

export const Default = template.bind({});

Default.args = defaultArgs;

export const Variants = (args) => ({
    components: {Dropdown},
    setup: () => ({args, variants}),
    template: '<Dropdown v-for="v of variants" class="me-1 mb-1 d-inline-block" :variant="v" v-bind="args" />',
});

Variants.args = defaultArgs;

export const Sizing = (args) => ({
    components: {Dropdown},
    setup: () => ({args, sizes}),
    template: '<Dropdown v-for="s of sizes" class="me-1 mb-1 d-inline-block" :size="s" v-bind="args" />',
});

Sizing.args = defaultArgs;

export const Menu = (args) => ({
    components: {DropdownMenu},
    setup: () => ({args}),
    template: '<DropdownMenu class="d-block" v-bind="args" />',
});

Menu.args = {
    items: [
        {label: 'Dropdown header', header: true},
        {label: 'Item one'},
        {label: 'Item two'},
        {label: 'Active item', active: true},
        {label: 'Item three'},
        {divider: true},
        {label: 'Item four'},
        {label: 'Item five'},
    ],
};
