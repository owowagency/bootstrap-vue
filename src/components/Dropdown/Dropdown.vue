<template>
    <div
        ref="dropdown"
        class="dropdown"
    >
        <!-- @slot Displays the dropdown toggle -->
        <slot name="dropdownToggle">
            <Btn
                class="dropdown-toggle"
                :class="toggleClass"
                data-bs-toggle="dropdown"
                :size="size"
                :variant="variant"
            >
                <!-- @slot Displays content inside the dropdown toggle -->
                <slot>
                    Select
                </slot>
            </Btn>
        </slot>

        <!--
        @slot Displays the dropdown menu
        @binding {array} items The `Item` list
        -->
        <slot
            name="dropdownMenu"
            v-bind="{'onClick:item': $attrs['onClick:item']}"
            :items="items"
        >
            <DropdownMenu
                :items="items"
                :class="menuClass"
                :item-class="itemClass"
                v-bind="{'onClick:item': $attrs['onClick:item']}"
            >
                <!-- Rollup does not dynamically overriding child slots. See issue #35
                <template
                    v-for="slotName in menuSlots"
                    #[slotName]="slotScope"
                >
                    <slot
                        :name="slotName"
                        v-bind="slotScope"
                    />
                </template> -->

                <template #prepend="slotScope">
                    <!-- @slot *Forwarded to child* -->
                    <slot
                        name="prepend"
                        v-bind="slotScope"
                    />
                </template>

                <template #items="slotScope">
                    <!-- @slot *Forwarded to child* -->
                    <slot
                        name="items"
                        v-bind="slotScope"
                    />
                </template>

                <template #item="slotScope">
                    <!-- @slot *Forwarded to child* -->
                    <slot
                        name="item"
                        v-bind="slotScope"
                    />
                </template>

                <template #append="slotScope">
                    <!-- @slot *Forwarded to child* -->
                    <slot
                        name="append"
                        v-bind="slotScope"
                    />
                </template>
            </DropdownMenu>
        </slot>
    </div>
</template>

<script lang="ts">
import Btn from '@/components/Button';
import DropdownMenu from '@/components/DropdownMenu';
import {dropdownItemProps} from '@/composables/useDropdownItem';
import {dropdownProps} from '@/composables/useDropdown';
import {ref} from 'vue';
import useBootstrapEmits from '@/composables/useBootstrapEmits';
import useBootstrapInstance from '@/composables/useBootstrapInstance';

const dropdownEvents = ['show', 'shown', 'hide', 'hidden'] as const;

type DropdownEvent = typeof dropdownEvents[number];
</script>

<script lang="ts" setup>

defineProps({
    ...dropdownItemProps,
    ...dropdownProps,
});

const emit = defineEmits<{(event: DropdownEvent): void}>();

const dropdown = ref<HTMLElement>();

useBootstrapEmits(
    dropdown,
    dropdownEvents,
    emit,
    'dropdown',
);

const {bsInstance: bsDropdown} = useBootstrapInstance(
    'Dropdown',
    dropdown,
);

// Rollup does not like dynamically overriding slots so this is not used for now.
// eslint-disable-next-line @typescript-eslint/no-unused-vars
const menuSlots = [
    'prepend',
    'items',
    'item',
    'append',
];

defineExpose({bsDropdown});
</script>

<docs>
```vue
<Dropdown :items="[Item]" />
```

## Imported Props

| Prop name | Description                                    | Type                                             | Values | Default   |
| --------- | ---------------------------------------------- | ------------------------------------------------ | ------ | --------- |
| items     | The items of the dropdown menu                 | [Item[]](../../composables/useDropdownItems)     | -      | () => []  |
| menuClass | The classes that will be forwarded to the menu | `string`\|`array`\|`object`                      | -      | undefined |
| size      | The size of the component                      | [Size](../../composables/useSize) (string)       | -      | 'md'      |
| variant   | The variant of the component                   | [Variant](../../composables/useVariant) (string) | -      | 'primary' |

<!-- TODO Auto-generate based on type? -->
## Events

| Name   | Args | Description                                                                                        |
| ------ | ---- | -------------------------------------------------------------------------------------------------- |
| show   |      | Fires immediately when the `show` instance method is called                                        |
| shown  |      | Fired when the dropdown has been made visible to the user and CSS transitions have completed       |
| hide   |      | Fires immediately when the `hide` instance method has been called                                  |
| hidden |      | Fired when the dropdown has finished being hidden from the user and CSS transitions have completed |
</docs>
