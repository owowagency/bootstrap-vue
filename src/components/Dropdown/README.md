# Dropdown

```vue
<Dropdown :items="[Item]" />
```

## Imported Props

| Prop name | Description                                    | Type                                             | Values | Default   |
| --------- | ---------------------------------------------- | ------------------------------------------------ | ------ | --------- |
| items     | The items of the dropdown menu                 | [Item[]](../../composables/useDropdownItems)     | -      | () => []  |
| menuClass | The classes that will be forwarded to the menu | `string`\|`array`\|`object`                      | -      | undefined |
| size      | The size of the component                      | [Size](../../composables/useSize) (string)       | -      | 'md'      |
| variant   | The variant of the component                   | [Variant](../../composables/useVariant) (string) | -      | 'primary' |

<!-- TODO Auto-generate based on type? -->

## Events

| Name   | Args | Description                                                                                        |
| ------ | ---- | -------------------------------------------------------------------------------------------------- |
| show   |      | Fires immediately when the `show` instance method is called                                        |
| shown  |      | Fired when the dropdown has been made visible to the user and CSS transitions have completed       |
| hide   |      | Fires immediately when the `hide` instance method has been called                                  |
| hidden |      | Fired when the dropdown has finished being hidden from the user and CSS transitions have completed |

## Slots

| Name           | Description                                 | Bindings                                 |
| -------------- | ------------------------------------------- | ---------------------------------------- |
| dropdownToggle | Displays the dropdown toggle                |                                          |
| default        | Displays content inside the dropdown toggle |                                          |
| dropdownMenu   | Displays the dropdown menu                  | <br/>**items** `array` - The `Item` list |
| prepend        | _Forwarded to child_                        |                                          |
| items          | _Forwarded to child_                        |                                          |
| item           | _Forwarded to child_                        |                                          |
| append         | _Forwarded to child_                        |                                          |
