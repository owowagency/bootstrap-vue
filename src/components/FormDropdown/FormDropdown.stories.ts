import FormDropdown from '.';
import {ref} from 'vue';
import {sizes} from '@/composables/useSize';

export default {
    title: 'Forms/Form dropdown',
    component: FormDropdown,
    argTypes: {},
};

const defaultArgs = {
    items: [
        {label: 'Item 1'},
        {label: 'Item 2'},
        {label: 'Item 3'},
    ],
};

const template = (args) => ({
    components: {FormDropdown},
    setup() {
        const value = ref();

        return {args, value};
    },
    template: `
        <FormDropdown
            v-model="value"
            v-bind="args"
        />
    `,
});

export const Default = template.bind({});

Default.args = defaultArgs;

export const Disabled = template.bind({});

Disabled.args = {
    disabled: true,
    ...defaultArgs,
};

export const Sizing = (args) => ({
    components: {FormDropdown},
    setup: () => ({args, sizes}),
    template: '<FormDropdown v-for="s of sizes" class="mb-1" :size="s" v-bind="args" />',
});

Sizing.args = defaultArgs;
