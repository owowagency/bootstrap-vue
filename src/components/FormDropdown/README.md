# FormDropdown

```vue
<FormDropdown :items="[Item]" />
```

## Imported Props

| Prop name | Description                                    | Type                                             | Values | Default   |
| --------- | ---------------------------------------------- | ------------------------------------------------ | ------ | --------- |
| items     | The items of the dropdown menu                 | [Item[]](../../composables/useDropdownItems)     | -      | () => []  |
| menuClass | The classes that will be forwarded to the menu | `string`\|`array`\|`object`                      | -      | undefined |
| size      | The size of the component                      | [Size](../../composables/useSize) (string)       | -      | 'md'      |
| variant   | The variant of the component                   | [Variant](../../composables/useVariant) (string) | -      | 'primary' |

## Props

| Prop name   | Description                                             | Type         | Values | Default   |
| ----------- | ------------------------------------------------------- | ------------ | ------ | --------- |
| disabled    | Disables the component                                  | disabledProp | -      |           |
| labelKey    | The key that is used to get the label from a given item | string       | -      | 'label'   |
| menuClass   | The classes that will be forwarded to the menu          | string       | -      | 'w-100'   |
| modelValue  | The model value                                         | object       | -      | undefined |
| placeholder | The placeholder of the dropdown                         | string       | -      | 'Select'  |
| size        |                                                         | sizeProp     | -      |           |
| toggleClass | The classes that will be forwarded to the toggle        | string       | -      | undefined |

## Slots

| Name                | Description                                | Bindings |
| ------------------- | ------------------------------------------ | -------- |
| dropdownToggle      | Displays the dropdown toggle               |          |
| dropdownToggleLabel | Displays the dropdown toggle label         | <br/>    |
| noOptions           | Displays content when there are no options |          |
| prepend             | Prepends the dropdown options              |          |
| items               | _Forwarded to child_                       |          |
| item                | _Forwarded to child_                       |          |
| append              | _Forwarded to child_                       |          |

## Events

| Event name        | Properties                      | Description                           |
| ----------------- | ------------------------------- | ------------------------------------- |
| update:modelValue | **value** `any` - The new value | Fired when the model value is updated |
