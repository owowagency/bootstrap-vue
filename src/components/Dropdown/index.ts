import Dropdown from './Dropdown.vue';

export default Dropdown;
