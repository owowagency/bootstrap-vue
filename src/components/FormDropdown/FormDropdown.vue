<template>
    <Dropdown
        class="form-dropdown"
        v-bind="propsDropdown"
        @click:item="emit('update:modelValue', $event)"
    >
        <template #dropdownToggle>
            <!-- @slot Displays the dropdown toggle -->
            <slot name="dropdownToggle">
                <div
                    class="form-select"
                    data-bs-toggle="dropdown"
                    :class="classes"
                >
                    <!-- @slot Displays the dropdown toggle label -->
                    <!-- @binding {{label: string, modelValue: Object}} dataset Bootstrap data attributes required for controlling the carousel -->
                    <slot
                        name="dropdownToggleLabel"
                        v-bind="{label, modelValue}"
                    >
                        {{ label }}
                    </slot>
                </div>
            </slot>
        </template>

        <template #prepend>
            <!-- @slot Displays content when there are no options -->
            <slot
                v-if="items.length === 0"
                name="noOptions"
            >
                <DropdownMenuItem
                    class="pe-none"
                    label="No options"
                />
            </slot>

            <!-- @slot Prepends the dropdown options -->
            <slot name="prepend" />
        </template>

        <!-- Rollup does not dynamically overriding child slots. See issue #35
        <template
            v-for="slotName in dropdownMenuSlots"
            #[slotName]="slotScope"
        >
            <slot
                :name="slotName"
                v-bind="slotScope"
            />
        </template> -->

        <template #items="slotScope">
            <!-- @slot *Forwarded to child* -->
            <slot
                name="items"
                v-bind="slotScope"
            />
        </template>

        <template #item="slotScope">
            <!-- @slot *Forwarded to child* -->
            <slot
                name="item"
                v-bind="slotScope"
            />
        </template>

        <template #append="slotScope">
            <!-- @slot *Forwarded to child* -->
            <slot
                name="append"
                v-bind="slotScope"
            />
        </template>
    </Dropdown>
</template>

<script lang="ts">
import useFormSelect, {formSelectProps} from '@/composables/useFormSelect';
import Dropdown from '@/components/Dropdown';
import DropdownMenuItem from '@/components/DropdownMenuItem';
import {computed} from 'vue';
import {dropdownProps} from '@/composables/useDropdown';
import extractKeysFrom from '@/library/extractKeysFrom';
import useClasses from '@/composables/useClasses';

const {disabled: disabledProp, size: sizeProp} = formSelectProps;
</script>

<script lang="ts" setup>
const props = defineProps({
    ...dropdownProps,
    /**
     * Disables the component
     */
    disabled: disabledProp,
    /**
     * The key that is used to get the label from a given item
     */
    labelKey: {
        type: String,
        default: 'label',
    },
    /**
     * The classes that will be forwarded to the menu
     */
    menuClass: {
        type: String,
        default: 'w-100',
    },
    /**
     * The model value
     */
    modelValue: {
        type: Object,
        default: undefined,
    },
    /**
     * The placeholder of the dropdown
     */
    placeholder: {
        type: String,
        default: 'Select',
    },
    size: sizeProp,
    /**
     * The classes that will be forwarded to the toggle
     */
    toggleClass: {
        type: String,
        default: undefined,
    },
});

const emit = defineEmits([
    /**
     * Fired when the model value is updated
     * @param {any} value The new value
     */
    'update:modelValue',
]);

const propsDropdown = extractKeysFrom(Object.keys(dropdownProps), props);

const label = computed(() => {
    if (!props.modelValue) {
        return props.placeholder;
    }

    return props.modelValue[props.labelKey];
});

const {classes} = useClasses(computed(() => [
    ...useFormSelect(props).classes.value,
    props.toggleClass,
]));

// Rollup does not like dynamically overriding slots so this is not used for now.
// eslint-disable-next-line @typescript-eslint/no-unused-vars
const dropdownMenuSlots = [
    'items',
    'item',
    'append',
];
</script>

<docs>
```vue
<FormDropdown :items="[Item]" />
```

## Imported Props

| Prop name | Description                                    | Type                                             | Values | Default   |
| --------- | ---------------------------------------------- | ------------------------------------------------ | ------ | --------- |
| items     | The items of the dropdown menu                 | [Item[]](../../composables/useDropdownItems)     | -      | () => []  |
| menuClass | The classes that will be forwarded to the menu | `string`\|`array`\|`object`                      | -      | undefined |
| size      | The size of the component                      | [Size](../../composables/useSize) (string)       | -      | 'md'      |
| variant   | The variant of the component                   | [Variant](../../composables/useVariant) (string) | -      | 'primary' |
</docs>
