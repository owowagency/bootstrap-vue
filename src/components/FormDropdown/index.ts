import FormDropdown from './FormDropdown.vue';

export default FormDropdown;
