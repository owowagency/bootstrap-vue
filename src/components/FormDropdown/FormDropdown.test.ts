import {describe, expect, it} from 'vitest';
import {mount, shallowMount} from '@vue/test-utils';
import Dropdown from '@/components/Dropdown';
import FormDropdown from '.';

const items = [
    {label: 'Item 1'},
    {label: 'Item 2'},
    {label: 'Item 3'},
];

describe('template', () => {
    componentRenderTest(FormDropdown, {props: {items}}, false);

    componentRenderTest(
        FormDropdown,
        {props: {items: []}},
        false,
        'renders no options',
    );

    componentRenderTest(
        FormDropdown,
        {props: {searchable: true}},
        false,
        'renders input',
    );

    componentSlotRenderTest(FormDropdown, 'dropdownToggle', {shallow: false});

    componentSlotRenderTest(FormDropdown, 'dropdownToggleLabel', {shallow: false});

    componentSlotRenderTest(FormDropdown, 'prepend', {shallow: false});

    componentSlotRenderTest(FormDropdown, 'items', {shallow: false});

    componentSlotRenderTest(FormDropdown, 'item', {
        props: {items},
        shallow: false,
    });

    componentSlotRenderTest(FormDropdown, 'append', {shallow: false});

    componentSlotRenderTest(
        FormDropdown,
        'noOptions',
        {
            shallow: false,
            props: {items: []},
        },
    );

    it('emits update:modelValue on click item', () => {
        const wrapper = shallowMount(FormDropdown);

        wrapper.findComponent(Dropdown).vm.$emit('click:item', 'yooo');

        expect(wrapper.emitted('update:modelValue')[0]).toEqual(['yooo']);
    });

    it('adds disabled class to .form-select', () => {
        const wrapper = mount(FormDropdown, {props: {disabled: true}});

        const formSelect = wrapper.find('.form-select');

        expect(formSelect.classes()).toContain('disabled');
    });

    it('adds size class to .form-select', () => {
        const wrapper = mount(FormDropdown, {props: {size: 'lg'}});

        const formSelect = wrapper.find('.form-select');

        expect(formSelect.classes()).toContain('form-select-lg');
    });

    it('adds toggle class to .form-select', () => {
        const wrapper = mount(FormDropdown, {props: {toggleClass: 'is-invalid'}});

        const formSelect = wrapper.find('.form-select');

        expect(formSelect.classes()).toContain('is-invalid');
    });
});

describe('label', () => {
    it('uses placeholder when no modelValue', () => {
        const wrapper = shallowMount(FormDropdown, {
            props: {placeholder: 'I am a placeholder'},
        });

        expect(wrapper.vm.label).toBe('I am a placeholder');
    });

    it('uses labelKey of modelValue when modelValue is given', () => {
        const wrapper = shallowMount(FormDropdown, {
            props: {
                modelValue: {name: 'Dees Dohmen'},
                labelKey: 'name',
            },
        });

        expect(wrapper.vm.label).toBe('Dees Dohmen');
    });
});
