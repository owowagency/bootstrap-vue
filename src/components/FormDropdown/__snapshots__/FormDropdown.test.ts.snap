// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders default 1`] = `
"<div class=\\"dropdown form-dropdown\\">
  <!-- @slot Displays the dropdown toggle -->
  <!-- @slot Displays the dropdown toggle -->
  <div class=\\"form-select form-select-md\\" data-bs-toggle=\\"dropdown\\">
    <!-- @slot Displays the dropdown toggle label -->
    <!-- @binding {{label: string, modelValue: Object}} dataset Bootstrap data attributes required for controlling the carousel -->Select
  </div>
  <!--
        @slot Displays the dropdown menu
        @binding {array} items The \`Item\` list
        -->
  <ul class=\\"dropdown-menu w-100\\">
    <!-- @slot Prepends the dropdown menu -->
    <!--
        @slot Displays content inside the dropdown menu
        @binding {array} items The \`Item\` list
        @binding {function} click-item The function to call when an item is clicked
        -->
    <!--
                @slot Displays the dropdown menu item
                @binding {object} item The \`Item\` object
                @binding {function} click-item The function to call when an item is clicked
                -->
    <li><a class=\\"dropdown-item\\" href=\\"#\\">Item 1</a></li>
    <!--
                @slot Displays the dropdown menu item
                @binding {object} item The \`Item\` object
                @binding {function} click-item The function to call when an item is clicked
                -->
    <li><a class=\\"dropdown-item\\" href=\\"#\\">Item 2</a></li>
    <!--
                @slot Displays the dropdown menu item
                @binding {object} item The \`Item\` object
                @binding {function} click-item The function to call when an item is clicked
                -->
    <li><a class=\\"dropdown-item\\" href=\\"#\\">Item 3</a></li><!-- @slot Appends the dropdown menu -->
  </ul>
</div>"
`;

exports[`template > renders input 1`] = `
"<div class=\\"dropdown form-dropdown\\" searchable=\\"true\\">
  <!-- @slot Displays the dropdown toggle -->
  <!-- @slot Displays the dropdown toggle -->
  <div class=\\"form-select form-select-md\\" data-bs-toggle=\\"dropdown\\">
    <!-- @slot Displays the dropdown toggle label -->
    <!-- @binding {{label: string, modelValue: Object}} dataset Bootstrap data attributes required for controlling the carousel -->Select
  </div>
  <!--
        @slot Displays the dropdown menu
        @binding {array} items The \`Item\` list
        -->
  <ul class=\\"dropdown-menu w-100\\">
    <!-- @slot Prepends the dropdown menu -->
    <!-- @slot *Forwarded to child* -->
    <!-- @slot Displays content when there are no options -->
    <li class=\\"pe-none\\"><a class=\\"dropdown-item\\" href=\\"#\\">No options</a></li><!-- @slot Prepends the dropdown options -->
    <!--
        @slot Displays content inside the dropdown menu
        @binding {array} items The \`Item\` list
        @binding {function} click-item The function to call when an item is clicked
        -->
    <!-- @slot Appends the dropdown menu -->
  </ul>
</div>"
`;

exports[`template > renders no options 1`] = `
"<div class=\\"dropdown form-dropdown\\">
  <!-- @slot Displays the dropdown toggle -->
  <!-- @slot Displays the dropdown toggle -->
  <div class=\\"form-select form-select-md\\" data-bs-toggle=\\"dropdown\\">
    <!-- @slot Displays the dropdown toggle label -->
    <!-- @binding {{label: string, modelValue: Object}} dataset Bootstrap data attributes required for controlling the carousel -->Select
  </div>
  <!--
        @slot Displays the dropdown menu
        @binding {array} items The \`Item\` list
        -->
  <ul class=\\"dropdown-menu w-100\\">
    <!-- @slot Prepends the dropdown menu -->
    <!-- @slot *Forwarded to child* -->
    <!-- @slot Displays content when there are no options -->
    <li class=\\"pe-none\\"><a class=\\"dropdown-item\\" href=\\"#\\">No options</a></li><!-- @slot Prepends the dropdown options -->
    <!--
        @slot Displays content inside the dropdown menu
        @binding {array} items The \`Item\` list
        @binding {function} click-item The function to call when an item is clicked
        -->
    <!-- @slot Appends the dropdown menu -->
  </ul>
</div>"
`;
